src/main/cpp/blaze_util.cc

```diff
@@ -74,6 +74,36 @@ bool GetNullaryOption(const char *arg, const char *key) {
   return true;
 }
 
+std::vector<const char*> SearchNaryOption(const vector<string>& args,
```

Comment:
It looks kind of dangerous to return vector of pointers as lifetime of provided buffers are bound by lifetime of **args**.
For example, if _SearchNaryOption_ is called with temporary argument for example (like you do in the tests)
returned vector will have pointers to freed memory.

I thing returning _vector<std::string>_ is better here. From memory copy perspective it would be almost the same. As
calling _FindLegacyUserBazelrc_ does implicit conversion to string from const char* now in code

Reply:
Thanks, @noxiouz! Adjusted to use vector instead.